api/src/services/sso-login.js

```diff
@@ -122,29 +115,19 @@ const getAuthorizationUrl = async (redirectUrl) => {
  * @returns {object} Token id and user details.
  */
 const getIdToken = async (currentUrl) => {
+  const serverConfig = await oidcServerSConfig();
+  const authorizationCodeGrant = () => client.authorizationCodeGrant(
+    serverConfig,
+    currentUrl,
+    { idTokenExpected: true }
+  );
+  const tokens = await authServerCallRetry(authorizationCodeGrant);
+  const { preferred_username, locale } = tokens.claims();
```

Comment:
One big missing piece I found was with the locale handling.  For a normal login, the user selects the language on the login page and this is passed in when we post the user's credentials.  However, we cannot use that selection for the SSO login because of how we get redirected through the OIDC provider.  Instead, I have added support here for loading the user's `locale` claim provided with the id_token.  So, you can specify the user's default locale in Keycloak/Entra and that will be what is used when logging into the CHT (regardless of what the user selects on the initial CHT login page).  (And, of course, the user can always change their locale from within the app after logging in.)

Reply:
Thank you for handling this.